File: problem0002/c/problem0002.c
/**
 * Problem: http://projecteuler.net/index.php?section=problems&id=2
 * Author: Elie De Brauwer <elie @ de-brauwer.be>
 * License: Simplified BSD
 * Date: December 12, 2010
 */

#include <stdio.h>

int main()
{
    int fib_prev = 1;
    int fib_cur = 2;
    int sum = 2;
    while(fib_cur < 4000000)
    {
        int tmp = fib_cur;
        fib_cur += fib_prev;
        fib_prev = tmp;
        if (fib_cur % 2 == 0)
        {
            sum += fib_cur;
        }
    }
    printf("%d\n", sum);
    return 0;
}

File: problem0002/haskell/problem0002.hs
{-
Problem: http://projecteuler.net/index.php?section=problems&id=2
Author: Elie De Brauwer <elie @ de-brauwer.be>
License: Simplified BSD
Date: December 12, 2010
-}

-- Generate a Fibonacci sequence starting with 1,2,3,5,8,...
fibo :: [Integer]
fibo = 1 : 2 : zipWith (+) fibo (tail fibo)

-- Get the sum of all even value not exceeding 4 million.
prob2 :: Integer
prob2 = sum (filter even (takeWhile (<4000000) fibo))

